dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,911 @@
+[[dev-guide-build-inputs]]
+= Build Inputs
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[how-build-inputs-work]]
+== How Build Inputs Work
+
+A _build input_ provides source content for builds to operate on. There are
+several ways to provide source in {product-title}. In order of precedence:
+
+* xref:dockerfile-source[Inline Dockerfile definitions]
+* xref:image-source[Content extracted from existing images]
+* xref:source-code[Git repositories]
+* xref:binary-source[Binary inputs]
+* xref:using-secrets-during-build[Input secrets]
+* xref:using-external-artifacts[External artifacts]
+
+Different inputs can be combined into a single build. As the inline Dockerfile takes
+precedence, it can overwrite any other file named *_Dockerfile_* provided by
+another input. Binary input and Git repositories are mutually exclusive inputs.
+
+Input secrets are useful for when you do not want certain resources or
+credentials used during a build to be available in the final application image
+produced by the build. External artifacts outside of a source repository can
+also be used to pull in additional files during a build.
```

Comment:
@bparees I added "Input secrets" and "External artifacts" to the above list. Some questions:

1) Is it problematic to try adding these 2 to the list since it's supposed to be "In order of precedence"? Where would they fall, or is that not really helpful?

2) I reordered the topic's later section order to match the precedence list above. I assume we had Git Source first previously because it's the most common. Any issues w/ reordering?

3) I added this above paragraph to touch briefly on input secrets and external artifacts since they were new additions and the other 4 had gotten touched on in the preceding paragraph. Any issues w/ the summary?